### ISSUE TITLE
Release MetaSeg on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [kushalvyas/Fit-Pixels-Get-Labels_MetaSeg#1]
Hi @kushalvyas 🤗

I'm Niels and work as part of the open-source team at Hugging Face. I discovered your excellent work on "Fit Pixels, Get Labels: Meta-learned Implicit Networks for Image Segmentation" and saw the notes about the demo and 3D code/models coming soon on your GitHub repository (https://github.com/kushalvyas/MetaSeg).

The Hugging Face Hub (huggingface.co) is a great platform for researchers to publish their artifacts, such as pre-trained model checkpoints. This enables better visibility and discoverability for your work, as you can add rich metadata tags.

Once your MetaSeg models and the demo are ready for release, would you be interested in hosting them on https://huggingface.co/models? We can help you with the process!

Hosting on Hugging Face will give your models more visibility and enable better discoverability. We can add tags in the model cards (the relevant pipeline tag for MetaSeg would be "image-segmentation") so that people find the models easier, link them to your paper page (https://huggingface.co/papers/2510.04021), etc.

If you're interested, here's a guide for uploading models: [https://huggingface.co/docs/hub/models-uploading](https://huggingface.co/docs/hub/models-uploading). For custom PyTorch models, the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class simplifies adding `from_pretrained` and `push_to_hub` functionalities.

We can also explore building a demo for MetaSeg on [Spaces](https://huggingface.co/spaces), and we might be able to provide a ZeroGPU [grant](https://huggingface.co/docs/hub/en/spaces-gpus#community-gpu-grants) to give you A100 GPUs for free.

Let me know if you're interested or need any guidance when the models are ready!

Kind regards,

Niels

**Comment by kushalvyas** [comment]
Hi Neils,

Thanks for the appreciation and the invitation to deploy on hugging face. We'll definitely consider it and get back to you soon!

Best,
Kushal
